front-end/src/components/analysis-page/IntentDiagram.test.js

```diff
@@ -117,6 +122,112 @@ describe("IntentDiagram", () => {
+        renderComponent(props);
+        const dispatch = jest.spyOn(store, 'dispatch');
+        screen.getByTestId("leaf-q1").focus();
+        userEvent.keyboard("{Enter}");
+        expect(dispatch).toHaveBeenCalledWith({"payload": "q1", "type": "analyzeTranscript/goToIntentByQuestion",});
+    });
+
+    it("should dispatch goToIntentByQuestion with the leaf's question if a leaf is focused and not End of Conversation and Space is pressed", () => {
+        renderComponent(props);
+        const dispatch = jest.spyOn(store, 'dispatch');
+        screen.getByTestId("leaf-q1").focus();
+        userEvent.keyboard("{Space}");
+        expect(dispatch).toHaveBeenCalledWith({"payload": "q1", "type": "analyzeTranscript/goToIntentByQuestion",});
+    });
+
+    it("should not have click function if a leaf is End of Conversation", () => {
+        props.children = {
+            "END OF CONVERSATION": 100,
+        };
+        renderComponent(props);
+        expect(screen.getByTestId("leaf-END OF CONVERSATION")).not.toHaveAttribute("onClick");
+    });
+
+    it("should not dispatch goToIntentByQuestion if a leaf is focused and End of Conversation and Enter is pressed", () => {
+        props.children = {
+            "END OF CONVERSATION": 100,
+        };
+        renderComponent(props);
+        const dispatch = jest.spyOn(store, 'dispatch');
+        screen.getByTestId("leaf-END OF CONVERSATION").focus();
+        userEvent.keyboard("{Enter}");
+        expect(dispatch).not.toHaveBeenCalled();
+    });
+
+    it("should not dispatch goToIntentByQuestion if a leaf is focused and End of Conversation and Space is pressed", () => {
+        props.children = {
+            "END OF CONVERSATION": 100,
+        };
+        renderComponent(props);
+        const dispatch = jest.spyOn(store, 'dispatch');
+        screen.getByTestId("leaf-END OF CONVERSATION").focus();
+        userEvent.keyboard("{Space}");
+        expect(dispatch).not.toHaveBeenCalled();
+    });
+
+    it("should have proper hover styling on a leaf if it is not End of Conversation", () => {
+        renderComponent(props);
+        expect(screen.getByTestId("leaf-q1")).toHaveClass("hover:bg-green-100");
+    });
+
+    it("should have button role on a leaf if it is not End of Conversation", () => {
+        renderComponent(props);
+        expect(screen.getByTestId("leaf-q1")).toHaveAttribute("role", "button");
+    });
+
+    it("should have tabIndex=0 on a leaf if it is not End of Conversation", () => {
+        renderComponent(props);
+        expect(screen.getByTestId("leaf-q1")).toHaveAttribute("tabIndex", "0");
+    });
+
+    it("should have proper aria label on a leaf if it is not End of Conversation", () => {
+        renderComponent(props);
+        expect(screen.getByTestId("leaf-q1")).toHaveAttribute("aria-label", "Click to view intent: q1");
+    });
```

Comment:
These tests are not necessary but are definitely appreciated!

Reply:
they test the correctness of the extremely complex ternary logic 😎